### DISCUSSION TITLE
The connection to the server 127.0.0.1:32768 was refused - did you specify the right host or port?

### DISCUSSION THREAD

**Discussion by uditgaurav** [community/community#25391] [Actions]
<p><strong>What is happening?</strong></p>
<ol>
<li>I’m Using a KinD cluster in Github Action the flow of workflow yaml is:</li>
<li>Creating KinD cluster (Created successfully)</li>
<li>Getting Nodes via <code>kubectl get nodes</code>
</li>
</ol>
<pre><code class="lang-auto">NAME                 STATUS   ROLES    AGE   VERSION
kind-control-plane   Ready    master   68s   v1.17.0
</code></pre>
<ol start="4">
<li>Now I’m trying to use the cluster inside my personal Github action bypassing kubeconfig. It is showing <strong>connection error</strong>:</li>
</ol>
<pre><code class="lang-auto">exit status 1: The connection to the server 127.0.0.1:32768 was refused - did you specify the right host or port?
</code></pre>
<p><em>Did I need to do anything else?</em></p>
<p><strong>Workflow-yaml</strong></p>
<pre><code class="lang-auto">name: Push
on:
  push:
    branches: [ master ]
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Creating KinD cluster
      run: |
        GO111MODULE=on 
        curl -Lo ./kind https://github.com/kubernetes sigs/kind/releases/download/v0.7.0/kind-$(uname)-amd64
        chmod +x ./kind
        sudo mv ./kind /usr/local/bin/kind
        kind version
        kind create cluster --loglevel debug
        sleep 60

    - name: Getting Nodes
      run: |
        kubectl get nodes

    - name: Set config
      run: echo ::set-env name=KUBE_CONFIG_DATA::$(base64 -w 0 ~/.kube/config)
  
      #This is the custom Github Action inside which the kubectl command is not running.
    - name: Running Litmus pod delete chaos experiment
      uses: uditgaurav/kubernetes-chaos@v0.1.0
</code></pre>

**Comment by chrispat** [comment]
<p>It looks like your action is implemented as a docker container, but you started your kind cluster on the daemon on the VM host.  Given the way docker networking is setup I don’t think you are going to be able to see that published port from inside a container.  Also, your action will not have access to ~/.kube/config as that is the home path on the VM.</p>
<p>If you run your tests using a bash script or an action implemented in JavaScript it should work.</p>

**Comment by uditgaurav** [comment, marked as answer]
<p>Yes, for this specific case<br>
We can use kind inside action using<br>
<code> kind get kubeconfig --internal &gt;$HOME/.kube/config</code><br>
it works for the version 0.7 of kind</p>
